include/mesh/mesh_inserter_iterator.h

```diff
@@ -39,11 +43,13 @@ class Point;
  * arguments, which adds objects to the Mesh.
  * Although any mesh_inserter_iterator can add any object, we
  * template it around object type so that type inference and
- * iterator_traits will work.
+ * iterator_traits will work. This class used to be intended
+ * for use with packed range methods in TIMPI. However, our
+ * packing routines now automatically add elements and nodes
+ * to the mesh, so this class is deprecated
```

Comment:
I would assume that it would just waste CPU cycles, but I feel like you would know the answer to this better than I... I figured our early return checks would do the job. Things like
```c++
Elem * DistributedMesh::add_elem (Elem * e)
{
  // Don't try to add nullptrs!                                                                                       
  libmesh_assert(e);

  // Trying to add an existing element is a no-op                                                                     
  if (e->valid_id() && _elements[e->id()] == e)
    return e;
```

Reply:
Yeah, I think you're probably correct here.  And if not, anyone savvy enough to use this API is probably able to keep up with the changes too.